core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.kt

```diff
@@ -0,0 +1,215 @@
+        mode: ActionMode,
+        item: MenuItem
+      ): Boolean {
+        if (item.itemId == id.menu_context_delete) {
+          dialogShower.show(DeleteSelectedHistory, {
+            historyViewModel.actions.offer(
+              DeleteHistoryItems(historyAdapter.items.filterIsInstance<HistoryItem>().filter { it.isSelected })
+            )
+            historyViewModel.actions.offer(Action.ExitActionModeMenu)
+            mode.finish()
+          })
+          return true
+        }
+        historyViewModel.actions.offer(Action.ExitActionModeMenu)
+        return false
+      }
+
+      override fun onDestroyActionMode(mode: ActionMode) {
+        historyViewModel.actions.offer(Action.ExitActionModeMenu)
+        historyViewModel.actions.offer(ToggleShowAllHistoryAvailability(history_switch))
+        actionMode = null
+      }
+    }
+
+  private val historyAdapter: HistoryAdapter2 by lazy {
+    HistoryAdapter2(
+      HistoryItemDelegate(this)
+    )
+  }
+
+  override fun injection(coreComponent: CoreComponent) {
+    activityComponent.inject(this)
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_history)
+    setSupportActionBar(toolbar)
+
+    supportActionBar?.setDisplayHomeAsUpEnabled(true)
+    supportActionBar?.setTitle(string.history)
+
+    recycler_view.adapter = historyAdapter
+    recycler_view.layoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
+
+    history_switch.setOnCheckedChangeListener { _, isChecked ->
+      historyViewModel.actions.offer(ToggleShowHistoryFromAllBooks(isChecked))
+    }
+    history_switch.isChecked = !sharedPreferenceUtil.showHistoryCurrentBook
+
+    compositeDisposable.add(historyViewModel.effects.subscribe { it.invokeWith(this) })
+  }
+
+  override fun onDestroy() {
+    compositeDisposable.clear()
+    super.onDestroy()
+  }
+
+  override fun onCreateOptionsMenu(menu: Menu): Boolean {
+    menuInflater.inflate(R.menu.menu_history, menu)
+    val search = menu.findItem(id.menu_history_search).actionView as SearchView
+    search.queryHint = getString(string.search_history)
+    search.setOnQueryTextListener(SimpleTextListener {
+      historyViewModel.actions.offer(Filter(it))
+    })
+    historyViewModel.state.observe(this, Observer(::render))
+    return true
+  }
+
+  private fun requestDeletionOfSelectedItems(){
+    dialogShower.show(DeleteSelectedHistory, {
+      historyViewModel.actions.offer(
+        DeleteHistoryItems(historyAdapter.items.filterIsInstance<HistoryItem>().filter { it.isSelected })
+      )
+      historyViewModel.actions.offer(Action.ExitActionModeMenu)
+    })
+  }
+
+  override fun onOptionsItemSelected(item:MenuItem) : Boolean {
+    if (item.itemId == android.R.id.home) {
+      historyViewModel.actions.offer(ExitHistory)
+      return true
+    }
+    if(item.itemId == R.id.menu_history_clear){
+      dialogShower.show(DeleteAllHistory, {
```

Comment:
This should be an action, ideally this file will not even have a DialogShower

```
historyAdapter.items
```

here you reach into the view to get data which should be avoided